src/Components/ServiceSelectionScene/SelectServiceButton.tsx

```diff
@@ -16,14 +16,61 @@ export interface SelectServiceButtonState extends SceneObjectState {
   labelValue: string;
   labelName: string;
 }
-export function selectLabel(primaryLabelName: string, primaryLabelValue: string, sceneRef: SceneObject) {
-  const variable = getLabelsVariable(sceneRef);
 
+export class SelectServiceButton extends SceneObjectBase<SelectServiceButtonState> {
+  public getLink = () => {
+    if (!this.state.labelValue) {
+      return;
+    }
+
+    return getLabelDrilldownLink(this.state.labelName, this.state.labelValue, this);
+  };
+
+  public onClick = () => {
+    selectLabel(this.state.labelName, this.state.labelValue, this);
+  };
+
+  public static Component = ({ model }: SceneComponentProps<SelectServiceButton>) => {
+    const styles = useStyles2(getStyles);
+    const labels = getLabelsVariable(model);
+    // Re-render links on label filter changes
+    labels.useState();
+    const link = model.getLink();
+    return (
+      <LinkButton
+        data-testid={testIds.index.showLogsButton}
+        tooltip={`View logs for ${model.state.labelValue}`}
+        className={styles.button}
+        variant="secondary"
+        size="sm"
+        disabled={!link}
+        href={model.getLink()}
+        onClick={model.onClick}
```

Comment:
It makes sense, but just FYI, this onClick is not triggered when opening in a new tab.

Reply:
I'm seeing it get called in the source window?